src/RileyMaze/userFunction.js

```diff
@@ -0,0 +1,182 @@
+/* exported RMZUserFunction */
+
+const {GObject, Gtk} = imports.gi;
+
+const {Codeview} = imports.codeview;
+
+// Parameters that we pass to the user functions while executing them.
+// Unlike the real parameters in the game, which will vary, these need to be
+// deterministic. If we're validating things before sending them to the game,
+// then we don't want something to be an error only sometimes. The game will
+// deal with runtime errors for that purpose.
+const COMMON_SCOPE = {
+    // Keep this in sync with UserScope class in
+    // hack-toy-apps/com.endlessm.RileyMaze/parameters.js
+    width: 1920,
+    height: 1080,
+    data: {},
+};
+
+const USER_FUNCTIONS = {
+    instructions: {
+        name: 'instructions',
+        args: [],
+        modelProp: 'instructionCode',
+        perLevel: true,
+        getScope() {
+            return Object.assign({}, COMMON_SCOPE);
```

Comment:
Here's where the `riley` object should go.

Reply:
Now that I think of it, we should probably put the `riley` object in its own scope (`INSTRUCTIONS_SCOPE`?) otherwise the Riley API will be available in the level editor too.